data_animation.py

```diff
@@ -0,0 +1,202 @@
+        self.line = self.axis.plot([], [], label=f"{display_name}", color=color)[0]
+
+    """a tick pushing the data forward one recording frame, and this might happen multiple 
+    times per animation frame"""
+    def tick(self):
+        self.plotted_data.append(self.data[time_cursor])
+        plotted_time_modified = [ts - config_ani.START_TIME_GRAPHICAL_OFFSET for ts in plotted_time]
+        self.line.set_data(plotted_time_modified, self.plotted_data)
+        return self.line
+
+
+# Import graphing settings from config
+unit_axies = {}
+
+for axis_name, axis_config in config_ani.AXIES.items():
+    unit_axies[axis_name] = PlotAxis(
+        axis_config["display_name"],
+        axis_config["unit"],
+        axis_config["min"],
+        axis_config["max"],
+        axis_config["primary"],
+    )
+
+lines = []
+
+for item_name, item_config in config_ani.PLOT_ITEMS.items():
+    lines.append(
+        PlotItem(
+            item_name,
+            item_config["csv_colum"],
+            unit_axies[item_config["axis"]],
+            item_config["color"],
+            item_config["filtered"],
+        )
+    )
+
+figure_title = config_ani.FIGURE_TITLE
+if config_ani.FIGURE_GRID:
+    plt.grid()
+plt.title(config_ani.FIGURE_TITLE, fontdict=config_ani.GLOBAL_FONT)
+
+# Generate one together legend for each of the curves
+handles = []
+labels = []
+for axis in unit_axies.values():
+    h, l = axis.axis.get_legend_handles_labels()
+    l_with_units = [f"{line} ({axis.unit})" for line in l]
+    handles += h
+    labels += l_with_units
+
+plt.legend(handles, labels, loc=config_ani.LEGEND_LOCATION, prop=config_ani.GLOBAL_FONT)
+
+
+# The method for this timing to work is that the animation runs at a certain framerate 
+# (converted to a time interval), and a time length
+# For each frame, it'll step up the running time variable by the time interval
+# It must then add all the data who's recorded timestamp is smaller than or equal to the
+# running time variable
+def animate(frame_number):
+    global time_cursor
+    global time
+    time += FRAME_INTERVAL / 1000  # adjusting to ms
+
+
+    # keep adding while not caught up
+    while (time_cursor < len(time_refference) and time_refference[time_cursor] <= time):  
+        # Add a recording frame to the ploted time, then to each value
+        plotted_time.append(time_refference[time_cursor])
+
+        for line in lines:
+            line.tick()
+
+        if time_cursor < len(time_refference): 
+            time_cursor += 1
+
+    updated_lines = [line.line for line in lines]
+
+    if frame_number % 100 == 0:
+        print(
+            f"Frame {frame_number+1} of {FRAME_LENGTH}" +\
+                f"({round((frame_number+1)/FRAME_LENGTH*100, 2)}%)        ",
+            end="\r",
+        )  # add 1 to frame counts to adjust for 0 index
+
+    return updated_lines
+
+
+SKIP_TO_FINAL = config_ani.SKIP_TO_FINAL
+PREVIEW = config_ani.PREVIEW
+
+if SKIP_TO_FINAL:
```

Comment:
Just a suggestion on code style, but there is a quote from the late, great, Jacob Deery

"Try and have as much of the code as possible in functions in other files, the main file should be as possible."

I think alot of the logic in this file is better placed in functions outside of the file, and potentially make a file called "main.py" which just imports and runs everything

Reply:
That's actually a good point. We should wrap all the code in a function and run it inside of a main.py

Reply:
Yep I agree. It's currently done like this because it's an evolution of a standalone script, but I'll do that now.

Reply:
I've given a shot at splitting it up into different files, but given the way this is laid out and how almost everything is interacting with matplotlib plt directly, or is just already importing lots of settings from other files, I think trying to split things out would be significantly more refactoring than I want to do right now.

Reply:
It's been broken into its own main.py script in [c928684](https://github.com/waterloo-rocketry/data-video-visualization/pull/1/commits/c9286849fd8801e7a57d8e0b004901f1a6766863)